packages/map/src/leaflet/Text.ts

```diff
@@ -0,0 +1,100 @@
+import { BeautifyIcon, Map } from "@hpcc-js/leaflet-shim";
+import { format as d3Format } from "d3-format";
+import { Markers } from "./Markers";
+
+export class Text extends Markers {
+
+    constructor(cluster = false) {
+        super(cluster);
+    }
+
+    layerUpdate(map: Map) {
+        const columns = this.columns();
+        const textIdx = columns.indexOf(this.textColumn());
+        const textColorIdx = columns.indexOf(this.textColorColumn());
+        const strokeColorIdx = columns.indexOf(this.strokeColorColumn());
+        const fillColorIdx = columns.indexOf(this.fillColorColumn());
+        super.layerUpdate(map, (row) => {
+            let text = this.propValue(textIdx, row, this.text());
+            const textSize = this.textSize(text, this.fontFamily(), this.fontSize());
+            if (this.textFormat_exists()) {
+                text = d3Format(this.textFormat())(text);
+            }
+            return {
+                icon: BeautifyIcon({
+                    iconShape: "rectangle",
+                    // isAlphaNumericIcon: true,
+                    // text: this.propValue(textIdx, row, this.text()),
+                    iconAnchor: [
+                        (textSize.width / 2) + this.textOffsetX(),
+                        (textSize.height / 2) + this.textOffsetY()
+                    ],
+                    html: `<span style="
+                        font-family:${this.fontFamily()};
+                        font-size:${this.fontSize()}px;
+                        pointer-events: none;
+                        white-space:nowrap;
+                        ">${text}</span>`,
+                    textColor: this.propValue(textColorIdx, row, this.textColor()),
+                    borderColor: this.propValue(strokeColorIdx, row, this.strokeColor()),
+                    backgroundColor: this.propValue(fillColorIdx, row, this.fillColor()),
+                    props: {
+                        owner: this,
+                        row
+                    }
+                }),
+                draggable: false
+            };
```

Comment:
Something similar could be achieved without the BeutifyIcon (and would be "lighter")?

Reply:
Wouldn't this require changing core classes for a very minor performance benefit? As it is this is very similar to `Pins.ts`.